mkosi/__init__.py

```diff
@@ -1412,6 +1414,24 @@ def build_kernel_modules_initrd(state: MkosiState, kver: str) -> Path:
     return kmods
 
 
+def join_initrds(initrds: Sequence[Path], output: Path) -> Path:
+    assert initrds
+
+    if len(initrds) == 1:
+        copy_tree(initrds[0], output)
+        return output
+
+    seq = []
+    for p in initrds:
+        initrd = p.read_bytes()
+        n = len(initrd)
+        padding = b'\0' * (round_up(n, 4) - n)  # pad to 32 bit alignment
+        seq += [initrd, padding]
+
+    output.write_bytes(b''.join(seq))
+    return output
```

Comment:
it doesn't have to be high performance and the usual case will probably be a single initrd, but this would still be nicer with a lot less copying around of the stuff by using [`io.BytesIO`](https://docs.python.org/3.11/library/io.html?highlight=bytesio#io.BytesIO), as a simple in memory file like buffer, that add the end produces a bytestring value that can be written out.